src/pages/scripting/reference/index.md

```diff
@@ -0,0 +1,172 @@
+- Coalesces all document changes into a single history state defined as `options.name`
+
+**Parameters**
+
+| Name | Type | Description |
+| --- | --- | --- |
+| options | Object | `documentID`: ID of the document whose history state should be suspended. |
+| | | `name`: name that is used for the history state - visible in the History panel|
+
+**`executionContext.hostControl.resumeHistory(suspensionID, commit)`**
+
+- Resumes the history state
+- Optionally creates a history state for the current document state
+
+**Parameters**
+
+| Name | Type | Description |
+| --- | --- | --- |
+| suspensionID | string | the suspension identifier that was returned from `suspendHistory` |
+| commit | bool | If true, then the current document state is committed and a history state is created. If false, the document is rolled back to the time when the state was suspended. (optional and defaults to true)|
+
+#### Example
+This example demonstrates suspending the history state on a target document, then resuming the state after modifying the document.
+```js
+let hostControl = executionContext.hostControl;
+// Get an ID for a target document
+let documentID = await getTargetDocument();
+
+// Suspend history state on the target document
+// This will coalesce all changes into a single history state called
+// 'Custom Command'
+let suspensionID = await hostControl.suspendHistory({
+    "documentID": documentID,
+    "name": "Custom Command"
+});
+
+// modify the document
+// . . .
+
+// resume the history state
+await hostControl.resumeHistory(suspensionID);
+```
+
+#### Automatic document closing
+**`executionContext.hostControl.registerAutoCloseDocument(docID)`**
+
+- Register a document to be automatically closed without saving when the execution context ends
+
+**`executionContext.hostControl.unregisterAutoCloseDocument(docID)`**
+
+- Unregister a document to be automatically closed without saving when the execution context ends
+
+**Parameters**
+
+| Name | Type | Description |
+| --- | --- | --- |
+| docID | number | The ID of the document of this history state 
+
+#### Example
+This example demonstrates creating and marking a document as "auto close" first. After adding some contents to the page, the document is unregistered from the set of auto close documents. If the user cancels while the script is running, the document is closed. 
+```js
+let hostControl = executionContext.hostControl;
+let docID = await createDocument();
+await hostControl.registerAutoCloseDocument(docID);
+
+// Add contents to docID
+...
+
+await hostControl.unregisterAutoCloseDocument(docID);
+```
+
+
+### Events
+
+**`onCancel`**
+
+Script execution can occur either when the user clicks “Cancel” in the progress bar, or when the host app encounters some exception in running the script file. Developers can add **event handlers** to get notified when the command has been cancelled. The associated callback will receive a parameter `reason`. 
```

Comment:
"Script execution" ?  Did you mean "cancellation"?

Reply:
good catch